src/pkg/util/nvidiautils/nvidiautils.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package nvidiautils
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// GetNvidiaBindPath returns a string array consisting of filepaths of nvidia
+// related files to be added to the BindPaths
+func GetNvidiaBindPath(abspath string) []string {
+	var strArray []string
+	var bindArray []string
+	var searchArray []string
+	var commentID = regexp.MustCompile(`#`)
+	var soID = regexp.MustCompile(".so")
+
+	// use nvidia-container-cli (if present)
+	command, err := exec.LookPath("nvidia-container-cli")
+	if err == nil {
```

Comment:
```
if err != nil {
    return nil, fmt.Errorf("nvidia-container-cli not found: %v", err)
}
```

Reply:
That's what I was thinking at first too @bauerm97 but I don't think that will work in this instance.  If singularity cannot find `nvidia-container-cli` or if `nvidia-container-cli` returns an error it should just issue a warning and then move on to try to use the conf file instead.

Reply:
Hmmm... nvidia-container-cli is an optional component. Don't want to return if error.

Reply:
Fair enough.

Reply:
In this case, I would actually say to break these into functions:

```
func nvidiaContainerCli() ([]string, error) {
    // use nvidia-container-cli (if present)
    command, err := exec.LookPath("nvidia-container-cli")
    if err =! nil {
        return nil, nil
    }
}      
```

Then within the main code can simply do this:

```
s, err := nvidiaContainerCli()
if err != nil {
    return nil, fmt.Println("Unable to do something...: %v", err)
}
...
```

Reply:
Ok, I like this better...